crates/nu-command/src/filesystem/util.rs

```diff
@@ -185,13 +184,38 @@ pub mod users {
         Gid::current().as_raw()
     }
 
+    #[cfg(not(any(target_os = "linux", target_os = "android")))]
     pub fn get_current_username() -> Option<String> {
         User::from_uid(Uid::current())
             .ok()
             .flatten()
             .map(|user| user.name)
     }
 
+    #[cfg(any(target_os = "linux", target_os = "android"))]
+    pub fn current_user_groups() -> Option<Vec<Gid>> {
+        // SAFETY:
+        // if first arg is 0 then it ignores second argument and returns number of groups present for given user.
+        let ngroups = unsafe { libc::getgroups(0, core::ptr::null::<gid_t> as *mut _) };
+        let mut buff: Vec<gid_t> = vec![0; ngroups as usize];
+
+        // SAFETY:
+        // buff is the size of ngroups and  getgroups reads max ngroups elements into buff
+        let found = unsafe { libc::getgroups(ngroups, buff.as_mut_ptr()) };
+
+        if found < 0 {
+            None
+        } else {
+            buff.truncate(ngroups as usize);
```

Comment:
Should we truncate to `found` here? If I read `man getgroups` correctly, this should also contain the length if given not `0` to `size`. Not sure if there could be a small TOCTOU risk here?
```suggestion
            buff.truncate(found as usize);
```

Reply:
yes, it make sense to use found instead of ngroups